src/irc/core/irc-servers-setup.c

```diff
@@ -97,7 +136,13 @@ static void sig_server_setup_fill_chatnet(IRC_SERVER_CONNECT_REC *conn,
 			    ircnet->sasl_password != NULL && *ircnet->sasl_password) {
 				conn->sasl_mechanism = SASL_MECHANISM_PLAIN;
 				conn->sasl_username = ircnet->sasl_username;
-				conn->sasl_password = ircnet->sasl_password;
+				char *password = NULL;
+				if (g_str_has_prefix(ircnet->sasl_password, "file:"))
+					read_password_file(ircnet->sasl_password, &password);
+				else
+					password = g_strdup(ircnet->sasl_password);
+				conn->sasl_password = g_strdup(password);
```

Comment:
You're introducing a g_strdup() here with no corresponding g_free().

This might be the wrong place to do this, why not when ircnet->sasl_password is set?